src/GitInfo/build/GitInfo.targets

```diff
@@ -339,7 +339,7 @@
 			<GitIsDirty>0</GitIsDirty>
 		</PropertyGroup>
 
-		<WriteLinesToFile File="$(_GitIsDirtyFile)" Lines="$(GitIsDirty)" WriteOnlyWhenDifferent="true"/>
+		<WriteLinesToFile File="$(_GitIsDirtyFile)" Lines="$(GitIsDirty)" Overwrite="true" WriteOnlyWhenDifferent="true"/>

```

Comment:
I believe this change makes this task append to the file once the file was created. `Overwrite` has to be `true` for `WriteOnlyWhenDifferent` to be used at all it seems:
https://github.com/dotnet/msbuild/blob/2e82c9f3f5feaa9444e67c4509c25f98fb1ea500/src/Tasks/FileIO/WriteLinesToFile.cs#L89-L95

Reply:
@bddckr: I'm not sure I follow your comment. The current behavior, in my own investigation of this issue, is that this task appends a new line containing the value of `($GitIsDirty)` to the file every time the task is called.

This change, by adding `Overwrite="true"`, modifies this task's behavior so that the file's contents get replaced if and only if:

1. the file already exists and
2. the file's contents differ from the value of `$(GitIsDirty)`;

and, otherwise, the task no-ops.

What happens when the file does not exist  - create it and write the contents of `$(GitIsDirty)` into it - is, to the best of my knowledge, unaffected by this change.

Am I missing something? I can vouch for this change fixing incremental builds for the projects I integrated `GitInfo` into.

Reply:
@AmoreCadenza Apologies, I must have read the diff as exactly the wrong way round, making me think you're breaking the functionality that you explained to be fixing 🤦

As we can see in the link to the source of the task that I posted earlier, your fix is basically what's turning on the `WriteOnlyWhenDifferent` behavior - because the source only ever even checks that property if `Overwrite` is also true.

tl;dr:
Before this change `WriteOnlyWhenDifferent` was never used as it's only ever used if `Overwrite` is turned on. With this change it is thus being used, ultimately giving the intended function of only writing _when needed_.

Consider this comment resolved! (Looks like I can't mark it as such, even though I'm the comment thread starter.)

Reply:
Thanks for clearing that up, @bddckr! I'm glad we're on the same page. Let me know if there's anything else needed to get this merged. 😄

I see a button to resolve this conversation thread on my end so I'll try pressing it.